Live metrics docs

tapadipti opened the pull request:
Added docs for live metrics.

Closes https://github.com/iterative/studio/issues/4285

Pending:

- [x] 1. Update screenshots
- [ ] ~2. Add some tutorial videos~

---

In review app: https://dvc-org-live-metrics-gjssweuin.herokuapp.com/doc/studio

daavoo commented on content/docs/studio/user-guide/projects-and-experiments/live-metrics.md:
```diff
@@ -0,0 +1,104 @@
+
+You should define the following environment variables in your CI job:
+
+- `STUDIO_ENDPOINT`: The Iterative Studio ingestion endpoint. Set this to
+  `https://studio.iterative.ai/api/live`
```
End users don't need to set this env var, the default value already is `https://studio.iterative.ai/api/live`. The env var was added for internal usage (i.e. post to local server)

```suggestion
```

tapadipti replied:
Thanks @daavoo

daavoo commented on content/docs/studio/user-guide/projects-and-experiments/live-metrics.md:
```diff
@@ -0,0 +1,104 @@
+
+- `STUDIO_ENDPOINT`: The Iterative Studio ingestion endpoint. Set this to
+  `https://studio.iterative.ai/api/live`
+- `STUDIO_REPO_URL`: URL for the Git repository connected to the project. E.g.,
+  `https://github.com/iterative/example-get-started`
```
We can check with @Suor on the expected/supported formats but on my local tests the full URL didn't work and this was the expected format:

```suggestion
- `STUDIO_REPO_URL`: URL for the Git repository connected to the project. E.g.,
  `github:iterative/example-get-started`
```

tapadipti replied:
👍

jorgeorpinel commented on content/docs/studio/get-started.md:
```diff
@@ -1,13 +1,15 @@
 
 https://www.youtube.com/watch?v=hKf4twg832g
 
+TODO: Add a video about live metrics tracking
```
For now maybe make these HTML comments if you're concerned about accidental merges 🙂 

```suggestion
<!-- TODO: Add a video about live metrics tracking -->
```

tapadipti replied:
@jorgeorpinel I changed this everywhere, but then reverted it. When reviewing the docs by running it locally, it became harder to review coz it was not clear which images/videos were going to be replaced (coz the TODO text didn't display - as expected). An accidental merge will cause more issues than this anyway 🙂

jorgeorpinel commented:
Links to core PRs and/or internal material for context would be good, I think.

jorgeorpinel commented on content/docs/studio/get-started.md:
```diff
@@ -105,6 +107,13 @@ You should now see that a project has been added in your dashboard.
+
+1. You can use your project to [submit new experiments] from Iterative Studio.
+
+2. For the experiments that you submit from Iterative Studio, you can [track
+   live metrics][live-metrics] by using [DVCLive].
```
```suggestion
2. For the experiments that you submit from Iterative Studio, you can [track
   live metrics] by using [DVCLive].
```

tapadipti replied:
The reason I used `[live-metrics]` was for consistency across all the pages. The actual text may be slightly different in different pages, but I wanted to use a consistent name for the link. In fact, I was trying to figure out if/how I can define `[live-metrics]` in one place and use it across multiple pages - is this possible?

jorgeorpinel replied:
Yes, naming ref links is most useful when reusing them in links with different words.

tapadipti replied:
@jorgeorpinel can I consider this resolved?

jorgeorpinel replied:
Feel free to mark things resolved when they are :) Resolving...

jorgeorpinel replied:
Feel free to mark things resolved when they are :) Resolving...

jorgeorpinel commented on content/docs/studio/get-started.md:
```diff
@@ -148,3 +157,5 @@ You should now see that a project has been added in your dashboard.
 [create a team]: /doc/studio/user-guide/teams
 [sign up for the **basic**, **teams** or **enterprise** plan]:
   /doc/studio/user-guide/change-team-plan-and-size
+[live-metrics]: /doc/studio/user-guide/projects-and-experiments/live-metrics
```
```suggestion
[track live metrics]: /doc/studio/user-guide/projects-and-experiments/live-metrics
```

tapadipti replied:
Pls see response to above comment.

tapadipti replied:
@jorgeorpinel can I consider this resolved?

jorgeorpinel reviewed (commented):
💅🏼 FYI this also works:

jorgeorpinel commented on content/docs/studio/user-guide/projects-and-experiments/live-metrics.md:
```diff
@@ -0,0 +1,102 @@
+You can regenerate the access token at any point. The old access token will no
+longer be authorized to send live metrics and plots to Iterative Studio.
+
+### Reset (delete) access token
+
+You can delete the access token when you no longer need it.
+
+The option to delete the access token is also available when you try to change
+your password. That is, you can reset all your access credentials (your password
```
Reset or delete?

tapadipti replied:
It's called reset in one place (change password modal) and delete in another (profile page) in Studio. That is why I used both the terms in the docs. But I guess it's better to fix this in Studio itself.
@mvshmakov can we change to one term everywhere - I think delete is clear, wdyt?

jorgeorpinel replied:
Reset sounds like something is preserved but returned to its original condition. I think we mean delete 🙂

tapadipti replied:
I think so too. cc @mvshmakov

mvshmakov replied:
Yes, within the flow of changing the password and flow of confirming the new password after resetting it, we have the option to delete an access token.

We have a correct label in a tooltip hint that states the token will be deleted. I've also renamed the "Reset access token" to "Delete access token" in both places in the app for clarity here: https://github.com/iterative/studio/pull/4800.

So, in the docs, we need to state "delete the access token".

tapadipti replied:
Now, we don't have a separate section for `Reset / delete access token`. And I've reviewed the text in this page for clarity.

jorgeorpinel commented on content/docs/studio/user-guide/projects-and-experiments/live-metrics.md:
```diff
@@ -0,0 +1,102 @@
+and the access token) at once. This is handy if you suspect that your account
+security may have been compromised.
+
+## Configure your model training CI job
+
+You should define the following environment variables in your CI job:
```
Feels like it needs more context, at least links, to what model training CI jobs are, what they do in Studio, how to zet them up, etc.

tapadipti replied:
The context is set in the first section of this page. I've revised it to make it more clear. Let me know what you think.

jorgeorpinel commented on content/docs/studio/user-guide/projects-and-experiments/live-metrics.md:
```diff
@@ -0,0 +1,102 @@
+
+## Send the updates
+
+Now, whenever you log your metrics or plots using [DVCLive], they will be
+automatically sent to Iterative Studio.
```
Can it be clearer? Idea
```suggestion
When the training CI job runs or whenever you log your metrics or plots using [DVCLive], they will be
automatically sent to Iterative Studio.
```

tapadipti replied:
I've changed it, but slightly differently from what you have suggested. PTAL.

jorgeorpinel commented on content/docs/studio/user-guide/projects-and-experiments/live-metrics.md:
```diff
@@ -0,0 +1,102 @@
+
+1. [Set up an access token](#set-up-an-access-token)
+2. [Configure your model training CI job](#configure-your-model-training-ci-job)
+3. [Send and view the updates](#send-the-updates)
```
**View live metrics** is a separate H2 now.

tapadipti replied:
Yes. But the 3 steps are about **sending** live updates. So I didn't include **View live metrics** as a separate step here. This list is not the table of contents in this page. But let me know if you think this must be changed.

jorgeorpinel commented on content/docs/studio/user-guide/projects-and-experiments/live-metrics.md:
```diff
@@ -0,0 +1,102 @@
+
+## View live metrics
+
+The live metrics are displayed alongside the corresponding experiment commits,
+and updates to the live metrics are highlighted in orange. The running
+experiments themselves are displayed with a `Running` indicator.
+
+TODO: Add a screenshot of the `Running` indicator
+
+The number of experiments with new updates to the live metrics values are
+displayed in the `Live` icon, which can also be used to filter and show only
+running experiments in the table.
+
+Live plots are displayed in the
```
Minor: terminology consistency
```suggestion
The live metrics are plotted alongside the corresponding experiment commits,
and updates to the live metrics are highlighted in orange. The running
experiments themselves are displayed with a `Running` indicator.

TODO: Add a screenshot of the `Running` indicator

The number of experiments with new updates to the live metrics values are
displayed in the `Live` icon, which can also be used to filter and show only
running experiments in the table.

Plots of the live metrics are displayed in the
```

tapadipti replied:
> The live metrics are **plotted** 

Not really. The metrics scalars are displayed in the project table in the experiment (commit) row. They aren't really plotted.

> Plots of the live metrics

This is also not entirely true. The live plots are updates to the **plots** files, which may or may not have the same values as the live **metrics**. @mvshmakov pls correct me if I'm wrong.

jorgeorpinel replied:
Oh OK. It's just unclear what "Live plots" are. I thought the history of live metrics was plotted, not that plots were updated in real time. These seem like different features but not well separated in [this page](https://dvc-org-live-metrics-bw8woadv9.herokuapp.com/doc/studio/user-guide/projects-and-experiments/live-metrics) so far. Probably adding image samples would help too.

tapadipti replied:
Got it. I've added `and plots` in all applicable places. Check out [this commit](https://github.com/iterative/dvc.org/pull/4012/commits/be466c550b19ef85ff638710ca849d9386fa22e3).

mvshmakov replied:
For clarity, we may call such experiments (as a table row) "live experiments". These experiments may contain "live metrics" and "live plots". IIRC, we may theoretically have a situation when there are no live metrics, but only live plots set up and wise-versa as well.

So, what do you think about such terminology @tapadipti?

jorgeorpinel commented on content/docs/studio/user-guide/projects-and-experiments/live-metrics.md:
```diff
@@ -0,0 +1,107 @@
+from Iterative Studio, the CI/CD setup in your Git repository gets invoked. If
+this setup includes a model training process, it will be triggered.
+
+In this model training CI action, you can use [DVCLive] to send live updates to
+metrics and plots back to Iterative Studio, without writing them to your Git
+repository. This requires a 3-step process:
```
Why do we want to send updates to Studio bypassing the Git repo. Shouldn't Git be the source of truth? Where will logs be stored, DVC remote? Thanks

tapadipti replied:
Addressed in https://github.com/iterative/dvc.org/pull/4012/commits/22e52337f839ef931e18531b66b56eefe4b5ebfb

jorgeorpinel replied:
> This will help to keep your Git commit history unpolluted, as you
can push only the final results of your experiments while being able to view all
intermediate results in Studio (in the form of live metrics and plots).

Still unclear in general where logs will be stored. Doesn't DVCLive use a directory in the repo normally?

tapadipti replied:
You can choose not to commit / push the logs to your repo. So my understanding is that when you are running experiments, you will push the live metrics and plots to Studio but won't commit / push the DVCLive output. @daavoo is this explanation correct?

daavoo replied:
Considering the MVP, I don't think there is a reason to say `This will help to keep your Git commit history unpolluted, as you can push only the final results of your experiments`. 

The MVP doesn't really help with keeping Git commit history clean and I would drop those referneces (at least until we complete the follow-up proposal about local experiments)

The MVP value is `being able to view all intermediate results in Studio` (I would add: `while the experiment is running).

tapadipti replied:
Updated.

tapadipti commented on content/docs/studio/user-guide/projects-and-experiments/live-metrics.md:
```diff
@@ -0,0 +1,107 @@
+
+When you
+[submit a new experiment](/doc/studio/user-guide/projects-and-experiments/run-experiments)
+from Iterative Studio, the CI/CD setup in your Git repository gets invoked. If
```
Addressed in https://github.com/iterative/dvc.org/pull/4012/commits/22e52337f839ef931e18531b66b56eefe4b5ebfb

jorgeorpinel commented on content/docs/studio/get-started.md:
```diff
@@ -1,13 +1,15 @@
 # Get Started with Iterative Studio
 
-The following video provides you a quick overview of how to use Iterative
+The following two videos provide you a quick overview of how to use Iterative
 Studio.
 
 > Note that we have renamed DVC Studio to Iterative Studio and Views to
 > Projects.
 
 https://www.youtube.com/watch?v=hKf4twg832g
 
+TODO: Add a video about live metrics and plots tracking
```
Not sure it's a good idea to put 2 videos at the top of the Get Started. In fact for DVC we hid all the GS videos; See [Get Started: Data Versioning](https://dvc.org/doc/start/data-management/data-versioning):

<img width="1009" alt="image" src="https://user-images.githubusercontent.com/1477535/199325133-5605cba0-10fd-4a10-b4c6-a53a57f17f39.png">

tapadipti replied:
Removed the second video. I don't think we need it here.

jorgeorpinel replied:
Sounds good, thanks.

jorgeorpinel commented on content/docs/studio/get-started.md:
```diff
@@ -105,6 +107,13 @@ You should now see that a project has been added in your dashboard.
 
 2. You can also make the project public by [sharing it on the web].
 
+## Submit experiments and track live metrics and plots
```
💅🏼 but this is going to be long in the CONTENT bar:

<img width="758" alt="image" src="https://user-images.githubusercontent.com/1477535/199326844-91dc9da7-63c3-4081-930e-fce8ff54cc83.png">

> Try to always sanity check changes on the review app, in this case https://dvc-org-live-metrics-bw8woadv9.herokuapp.com/doc/studio/get-started.

jorgeorpinel commented on content/docs/studio/how-it-works.md:
```diff
@@ -1,48 +1,58 @@
-repositories. It works very closely with your Git ecosystem.
+[Iterative Studio](https://studio.iterative.ai/) works very closely with your
+Git ecosystem. It parses the data, metrics, hyperparameters and model metadata
+that you add to your ML project Git repositories, and renders them on a web UI.
 
-This video illustrates how Iterative Studio works closely with your Git
-ecosystem.
+Additionally, you can [track metrics from your experiments in real
+time][live-metrics-and-plots] by pushing the metrics directly to Iterative
+Studio using [DVCLive].
```
This seems like a general description. Why does it now link specifically to the one specific feature but nothing else from the list in the first paragraph? I think we could keep it general here e.g. avoid the new paragraph, just add "...parses the data and metrics in real time, hyperparams..."

jorgeorpinel replied:
Thanks

jorgeorpinel commented on content/docs/studio/how-it-works.md:
```diff
@@ -1,48 +1,58 @@
+time][live-metrics-and-plots] by pushing the metrics directly to Iterative
+Studio using [DVCLive].
+
+The following two videos illustrate how Iterative Studio works with data from
+your ML projects and experiments.
```
Minor but I'm also not sure we want 2 video embeds one after another here.

We have something similar here: https://dvc.org/doc/user-guide/experiment-management/experiments-overview#work-with-dvc-experiments-from-a-gui and it's a bit strange.

jorgeorpinel commented on content/docs/studio/user-guide/projects-and-experiments/run-experiments.md:
```diff
@@ -79,24 +79,47 @@ on `Commit changes`.
-At this point, the new experiment appears in the project's experiment table. If
-you just committed to a new branch, then a new pull request will also have been
-created from the new branch to the base branch.
+## What happens after you submit a new experiment
```
Can we use a more descriptive title and maybe add some text and/or list with links to the sections below as a summary? (following existing Studio docs style). Rough idea:

```suggestion
## Follow the experiment results

- [Git commit and PRs]
- [Model training]
- [Live metrics and plots]
```

tapadipti replied:
Changed. PTAL.

jorgeorpinel replied:
Still seeing the same title?

tapadipti replied:
Yes, I didn't change the title as suggested. But I changed the content as per this feedback: `add some text and/or list with links to the sections below as a summary? (following existing Studio docs style).`

jorgeorpinel commented on content/docs/studio/user-guide/projects-and-experiments/run-experiments.md:
```diff
@@ -79,24 +79,47 @@ on `Commit changes`.
-created from the new branch to the base branch.
+## What happens after you submit a new experiment
+
+### Git commit (and pull request) are created
+
+The new experiment appears in the project's experiment table. If you had
+specified a new branch to commit the changes to, then a new pull request will
+also have been created from the new branch to the base branch.
```
So is an experiment created? Is a commit created? This may be confusing. I don't think we should call experiments commits until they are "made persistent" (but we should probably also avoid that term).

> Also the grammatical number is unclear i.e. is it "Git commit (or PR) is" -or- "Git commit(s) and PR(s) are" ?

tapadipti replied:
It's 1 Git commit and 0 or 1 PR. Not sure what's the best way to say this.

jorgeorpinel replied:
OK, the grammatical number is correct then (sounds a bit strange but it's a very minor observation, moving on...)

I see the text is clarified. Thanks, resolving.

jorgeorpinel commented on content/docs/studio/user-guide/projects-and-experiments/visualize-and-compare.md:
```diff
@@ -73,6 +72,15 @@ experiment, you can use the plots to compare them.
 
 ![](https://static.iterative.ai/img/studio/plots.png)
 
+### Live (real-time) plots
```
```suggestion
### Live plots (real time)
```

tapadipti replied:
Didn't change this. Let me know if you strongly feel about this.

jorgeorpinel replied:
If it's the main section about the feature and the feature is called "Live plots" I do think we should word it consistently (for SEO at least).

tapadipti replied:
It's not the main section about the feature, as there is a separate page for Live metrics and plots. But it's ok to remove `real-time` from the title. I've changed it.

jorgeorpinel commented on content/docs/studio/user-guide/projects-and-experiments/live-metrics-and-plots.md:
```diff
@@ -0,0 +1,111 @@
+
+## Set up an access token
+
+Iterative Studio uses access tokens to authorize [DVCLive] to send live updates
+to the metrics and plots.
+
+### Generate access token
+
+To set up the access token, open your user profile page. In the
+`Studio access token` section, click on `Generate new token`.
+
+### Regenerate access token
+
+You can regenerate the access token at any point. The old access token will no
```
The whole section can be summarized into a paragraph TBH. Something like (very crude)

> You can generate, regenerate, or delete your access tokens from the `Studio access token` section of your profile page.

Which makes me wonder if we even need the H2

_p.s. why do we call delete "reset"? Reset sounds more like "regenerate", no? Can we avoid the term?_

jorgeorpinel commented on content/docs/studio/user-guide/projects-and-experiments/live-metrics-and-plots.md:
```diff
@@ -0,0 +1,111 @@
+
+TODO: Replace this with a video tutorial for live metrics and plots
+
+## Set up an access token
+
+Iterative Studio uses access tokens to authorize [DVCLive] to send live updates
+to the metrics and plots.
+
+### Generate access token
+
+To set up the access token, open your user profile page. In the
+`Studio access token` section, click on `Generate new token`.
+
+### Regenerate access token
+
+You can regenerate the access token at any point. The old access token will no
```
A lot of structure here. Are these H3s necessary? There is one or 2 sentences under each one. ~~A numbered or bullet list would probably do, it may even be clearer.~~

jorgeorpinel replied:
UPDATE: The whole section can be summarized into a paragraph TBH, maybe a single sentence. Something like (very crude):

> You can generate, regenerate, or delete your access tokens from the `Studio access token` section of your profile page.

Which makes me wonder if we even need the H2.

jorgeorpinel commented on content/docs/studio/user-guide/projects-and-experiments/live-metrics-and-plots.md:
```diff
@@ -0,0 +1,111 @@
+You can regenerate the access token at any point. The old access token will no
+longer be authorized to send live metrics and plots to Iterative Studio.
+
+### Reset (delete) access token
```
p.s. why do we call delete "reset"? Reset sounds more like "regenerate" (or something else) IMO. May be confusing: can we avoid the term?

jorgeorpinel commented on content/docs/studio/user-guide/projects-and-experiments/live-metrics-and-plots.md:
```diff
@@ -0,0 +1,111 @@
+
+- `STUDIO_REPO_URL`: URL for the Git repository connected to the project. E.g.,
+  `github:iterative/example-get-started`
+- `STUDIO_TOKEN`: Your access token. The access token must be present in any
+  request that sends data to the Iterative Studio ingestion endpoint. Requests
+  with missing or incorrect access tokens are rejected with an appropriate HTTP
+  error code and error message.
```
We could even try to include the sentence about generating the token in here and completely avoid the previous H2 (**Set up an access token**).

tapadipti replied:
Didn't do this, coz I think the access token set up deserves explaining separately. But I've made that section (on access token setup) small, so it should be ok now.

jorgeorpinel commented on content/docs/studio/user-guide/projects-and-experiments/live-metrics-and-plots.md:
```diff
@@ -0,0 +1,111 @@
+  error code and error message.
+
+One way to provide the above values to your CI job is to set them as GitHub
+secrets.
+
+Here is an example GitHub action.
```
💅🏼 
```suggestion
secrets. For example:
```

tapadipti replied:
Didn't change this. Since the example is specific to GitHub, I think we should mention it.

jorgeorpinel commented on content/docs/studio/user-guide/projects-and-experiments/live-metrics-and-plots.md:
```diff
@@ -0,0 +1,111 @@
+TODO: provide link to an example snippet (such as
+[this](https://github.com/iterative/test-dvclive-studio/blob/086a51d76c7983f24c091e1b007820916aa75e7d/.github/workflows/test_live_metrics.yaml#L17-L19))
+
+## Send the updates
+
+In the training CI job (which has been configured as detailed above), whenever
+you log your metrics or plots using [DVCLive], they will be automatically sent
+to Iterative Studio.
+
+Iterative Studio stores the live metrics and plots data in its database.
```
OK this H2 is also very short haha. We can probably compress these 3 initial H2s into a single section. Not sure what to call it since it's both about the config and starting using DVCLive...

jorgeorpinel commented on content/docs/studio/user-guide/projects-and-experiments/live-metrics-and-plots.md:
```diff
@@ -0,0 +1,111 @@
+
+Iterative Studio stores the live metrics and plots data in its database.
+
+## View live metrics and plots
+
+In the project table, the live metrics are displayed next to the Git commit
+corresponding to the experiment. Updates to the live metrics are highlighted in
```
Should this section be linked from the intro?

> If we reduce the previous 3 H2s into one then there would be 2 links up there.

tapadipti replied:
Sorry, I didn't get you.

jorgeorpinel commented on content/docs/studio/user-guide/projects-and-experiments/live-metrics-and-plots.md:
```diff
@@ -0,0 +1,111 @@
+
+In the project table, the live metrics are displayed next to the Git commit
+corresponding to the experiment. Updates to the live metrics are highlighted in
+orange. The running experiments themselves are displayed with a `Running`
+indicator.
```
This should be intuitive. No need to worry about explaining things like this 🙂:

```suggestion
orange.
```

jorgeorpinel replied:
> p.s. if the point is to emphasize about experiments running live, you could use the image caption for that (once there's a screenshot below).

jorgeorpinel replied:
Thanks

jorgeorpinel commented on content/docs/studio/user-guide/projects-and-experiments/live-metrics-and-plots.md:
```diff
@@ -0,0 +1,111 @@
+Live plots are displayed in the
+[plots pane](/doc/studio/user-guide/projects-and-experiments/visualize-and-compare#how-to-generate-plots).
+
+An experiment can have one of the following statuses:
+
+- **Running** - Iterative Studio expects to receive live metrics and plots for
+  these experiments.
```
p.p.s. But I think this is probably enough info about the running state.

tapadipti replied:
Guess there's nothing to do here, right?

jorgeorpinel commented on content/docs/studio/user-guide/projects-and-experiments/live-metrics-and-plots.md:
```diff
@@ -0,0 +1,111 @@
+
+- **Running** - Iterative Studio expects to receive live metrics and plots for
+  these experiments.
+- **Completed** - Iterative Studio does expect to receive any more updates for
```
```suggestion

- **Completed** - Iterative Studio does not expect to receive any more updates for
```

tapadipti replied:
Changed. Thanks.

jorgeorpinel commented on content/docs/studio/user-guide/projects-and-experiments/live-metrics-and-plots.md:
```diff
@@ -0,0 +1,111 @@
+  no longer be displayed for the experiment. You are expected to persist the
+  final results in Git. TODO: confirm if this statement is correct.
+
+- **Stopped/Error** - These experiments stopped due to some error.
```
Can the error message/log be found somewhere?

tapadipti replied:
I don't think we have details of the errors. @mvshmakov is there any details we can provide here?

daavoo replied:
I believe the release will not include the concept of stopped/errored experiment, only completed

tapadipti replied:
Oh ok. In that case, I should remove this from the docs. 
@daavoo what happens in case of an error - does it show up in Studio as running or completed?
cc @mvshmakov

mvshmakov replied:
As @daavoo said already, we don't have a notion of an errored-out live experiment. If, say, the process on the remote machine shuts down, DVCLive just stops sending any events and we'll have a never-ending progressing live experiment.

Users are able to delete such experiments as it is usually obvious to them that the process failed. We've discussed errors on the previous live metrics sync and it was stated that it is unrealistic to expect DVCLive to handle all types of errors that may happen.

tapadipti replied:
ok, thanks @mvshmakov. I'll update the docs accordingly.

tapadipti replied:
I've updated it [here](https://github.com/iterative/dvc.org/pull/4012/commits/ea378ba5f6317a16b2d6f76a397774dcfa9336bb).

jorgeorpinel replied:
Thanks guys! Resolving, I guess.

jorgeorpinel reviewed (commented):
Looks good in general. I'd approve but it says DON'T MERGE YET so I don't want to cause an accidental merge. We're still focusing too much in documenting sort of literally a visual product (which should intuitive) but that's OK: we'll probably get used to avoiding that slowly.

tapadipti commented on content/docs/studio/user-guide/prepare-your-repositories.md:
```diff
@@ -6,26 +6,16 @@
+models, datasets, metrics and hyperparameters defined in your Git repositories.
+These values can be added to a project in two ways:
+
+1. **Initialize DVC in your repositories**: You can use [DVC](https://dvc.org/)
```
Changed

jorgeorpinel commented on content/docs/studio/user-guide/projects-and-experiments/run-experiments.md:
```diff
@@ -79,24 +79,55 @@ on `Commit changes`.
 ![](https://static.iterative.ai/img/studio/cml_report_icon.png)
+
+[dvclive]: /doc/dvclive
+[cml]: (https://cml.dev)
```
Fix broken link
```suggestion
[cml]: https://cml.dev
```

jorgeorpinel commented:
There are file conflicts @tapadipti ☝🏼 I took a look but wasn't sure which text versions to keep.

jorgeorpinel commented on content/docs/studio/get-started.md:
```diff
@@ -105,6 +105,13 @@ You should now see that a project has been added in your dashboard.
 
 2. You can also make the project public by [sharing it on the web].
 
+## Run experiments and get live updates
+
+1. You can use your project to [submit new experiments] from Iterative Studio.
```
Seems repetitive vs the previous section: we already mentioned "After creating a project, you can... submit new experiments"

Can we combine the sections instead?

tapadipti replied:
Have updated, but somewhat differently from your suggestion.

jorgeorpinel replied:
Thanks @tapadipti and sorry for the quite-late follow-up. I'll send another PR if there are any important suggestions...

jorgeorpinel commented on content/docs/studio/how-it-works.md:
```diff
@@ -1,48 +1,54 @@
-[Iterative Studio](https://studio.iterative.ai/) works with the data, metrics,
-hyperparameters and model metadata that you add to your ML project Git
-repositories. It works very closely with your Git ecosystem.
+[Iterative Studio](https://studio.iterative.ai/) works very closely with your
+Git ecosystem. It parses the data, metrics, hyperparameters and model metadata
+that you add to your ML project Git repositories, and renders them on a web UI.
```
This change is unrelated in this PR, but since it was updated, here's a suggestion because the current wording is unclear, esp. around the concept of metadata:

```suggestion
[Iterative Studio](https://studio.iterative.ai/) works very closely with your
Git ecosystem. It parses the metadata found in your ML project repositories to
present organized data, hyperparameters, models, and metrics on a web UI.
```

jorgeorpinel commented on content/docs/studio/how-it-works.md:
```diff
@@ -1,48 +1,54 @@
+[Iterative Studio](https://studio.iterative.ai/) works very closely with your
+Git ecosystem. It parses the data, metrics, hyperparameters and model metadata
+that you add to your ML project Git repositories, and renders them on a web UI.
+It also parses and displays live metrics and plots sent by [DVCLive]; these
+don't need to be pushed to the Git repositories.
```
I think this is included in the "and metrics" part. We didn't even mention DVC earlier in this paragraph so why mention DVCLive? Too specific. The wording I suggested above also prevents the need to clarify about not having to push to Git. So I'd remove it from here
```suggestion
```
But maybe we can add this note somewhere else further down... Let me 👀

tapadipti replied:
We need to mention this here coz otherwise it's not clear that live metrics are not "found in your ML project repositories".

jorgeorpinel commented on content/docs/studio/how-it-works.md:
```diff
@@ -1,48 +1,54 @@
 
-This video illustrates how Iterative Studio works closely with your Git
-ecosystem.
+The following video illustrate how Iterative Studio works with data from your ML
+projects and experiments.
 
 > Note that we have renamed DVC Studio to Iterative Studio and Views to
 > Projects.
 
 https://www.youtube.com/watch?v=5xM5az78Lrg
```
The video and it's purpose (being in this place of the doc) should be self-explanatory. Also, proper admonition:

```suggestion
https://www.youtube.com/watch?v=5xM5az78Lrg

<admon>

We have renamed DVC Studio to Iterative Studio and Views to Projects.

</admon>
```

tapadipti replied:
Am not making this change, coz we use this same format in 3 places where we have videos; this note applies to the videos specifically.

jorgeorpinel commented on content/docs/studio/how-it-works.md:
```diff
@@ -1,48 +1,54 @@
-## How your project data is saved
+## How you save your project data
 
 - Using [DVC] and [Git], you will push all your ML experiments to your GitHub,
   GitLab or Bitbucket repositories as Git commits.
```
💅🏼 
```suggestion
- Using [DVC] and [Git], you will push all your ML experiments to your GitHub,
  GitLab or Bitbucket repositories.
```

jorgeorpinel commented on content/docs/studio/how-it-works.md:
```diff
@@ -1,48 +1,54 @@
 - Using Iterative Studio, or using the command line interface (CLI) of [GTO],
   and possibly [MLEM], you will push all your ML model details to the Git
   repositories as Git commits and Git tags.
+- Using [DVCLive], you will push live metrics and plots from your Git-committed
+  experiments to Iterative Studio.
```
💅🏼 minor (just a bit shorter)
```suggestion
- Using [DVCLive], you will push live metrics and plots directly to Iterative
  Studio.
```

tapadipti replied:
I think it is important to mention that the experiments are committed to Git, coz we don't support live metrics for local experiments yet.

tapadipti replied:
> from your experiments running on a remote machine.

This is still confusing. We need the experiments to have been submitted from Studio and committed to Git. `Remote machine` could be any remote machine - for me, if I do some work in a cloud instance, then I'm working on a remote machine coz it's not my local machine.

I've updated the text, but differently as shown below:

`Using [DVCLive], you will push live metrics and plots from your experiments to Iterative Studio. Note that at the moment, live metrics are supported only for experiments that you [submit from the Iterative Studio UI].`

Let me know if it still doesn't make sense.

mvshmakov replied:
It makes more sense now, thanks!

jorgeorpinel reviewed (commented):
> But maybe we can add this note somewhere else further down... (re https://github.com/iterative/dvc.org/pull/4012#pullrequestreview-1218407058)

OK, I see it's already mentioned here (3rd commented text below) so this should be covered ✅ - just some edits:

jorgeorpinel commented on content/docs/studio/how-it-works.md:
```diff
@@ -1,48 +1,54 @@
   [add the metrics and hyperparameters to your Git repositories manually](/doc/studio/user-guide/projects-and-experiments/configure-a-project#custom-metrics-and-parameters).
 - Details of your ML models, including versions and stage assignments, are
   extracted from the Git commits and tags.
+- Live metrics and plots for experiments committed to Git are extracted from the
+  data you send using [DVCLive].
```
Wait. I thought live metrics/plots were not committed/pushed to Git? The mechanics start getting confusing here.

tapadipti replied:
The live metrics/plots aren't committed to Git. But the experiments are (there has to be a Git commit that triggers the experiment).

tapadipti replied:
IMO, `remote` is not enough, coz it does not imply that there is a Git commit for it. Users could interpret `remote` to mean any (cloud) machine that is not your laptop/desktop.

jorgeorpinel replied:
I see 2 issues here:

- Emphasize where experiments run/are saved (local vs remote) and what that means to the user from the PoV of the Studio UI
- Differentiate (or not) between experiments and commits (long-standing DVC discussion). I incline to call the first "experiments" and the latter "commits" or "project versions" (even if we clarify that they used to be exps.

tapadipti replied:
@jorgeorpinel We've now already merged the docs for live metrics for local experiments. Can you pls read that version and comment if you still see issues?

jorgeorpinel replied:
The only part that confuses me a little now is under https://dvc.org/doc/studio/how-it-works#how-iterative-studio-runs-experiments-and-model-actions:

> ### How Iterative Studio runs experiments...
>
> - When you [run new experiments](https://dvc.org/doc/studio/user-guide/projects-and-experiments/run-experiments) ... in Iterative Studio, it creates Git commits

So are experiments commits? (Could be what a reader wonders.)

tapadipti replied:
Yes, the experiments are pushed to Git, as Git commits and possibly PRs (if the change was submitted to a new branch).

jorgeorpinel commented on content/docs/studio/how-it-works.md:
```diff
@@ -53,6 +59,8 @@ https://www.youtube.com/watch?v=5xM5az78Lrg
 - You can set up your CI/CD actions (e.g. GitHub Actions) to run model training
   upon the creation of Git commits, tags or pull requests. You can use [CML] in
   your CI/CD actions for continuous machine learning.
+- You can use [DVCLive] in your CI/CD actions to [send live metrics of model
+  training to Iterative Studio][live-metrics-and-plots].
```
Hmmm, how are these things that Studio does? (This is under **How Iterative Studio runs ... model actions**.)

tapadipti replied:
I've merged this point with the previous one; it should be better now.

jorgeorpinel commented on content/docs/studio/user-guide/prepare-your-repositories.md:
```diff
@@ -6,26 +6,15 @@
-   version your code, data and models all within your Git repositories. Data
-   Version Control, or DVC, is a data and ML experiment management tool that
-   takes advantage of the existing engineering toolset that you're already
+Iterative Studio creates [projects][project] by identifying machine learning
```
```suggestion
Iterative Studio creates [projects] by identifying machine learning
```

jorgeorpinel commented on content/docs/studio/user-guide/prepare-your-repositories.md:
```diff
@@ -55,3 +44,4 @@ Datasets, metrics, and hyperparameters can be added to a project in two ways:
 
 [on project settings]:
   /doc/studio/user-guide/projects-and-experiments/configure-a-project#non-dvc-repositories
+[project]: /doc/studio/user-guide/projects-and-experiments/what-is-a-project
```
```suggestion
[projects]: /doc/studio/user-guide/projects-and-experiments/what-is-a-project
```

jorgeorpinel reviewed (commented):
🤷🏼 😃

jorgeorpinel commented on content/docs/studio/user-guide/prepare-your-repositories.md:
```diff
@@ -6,26 +6,15 @@
-   Version Control, or DVC, is a data and ML experiment management tool that
-   takes advantage of the existing engineering toolset that you're already
+Iterative Studio creates [projects][project] by identifying machine learning
+models, datasets, metrics and hyperparameters defined in your Git repositories.
+These values can be added to a project in two ways:
```
Minor: Does the order matter here? I'd always go with "data, params, models, and metrics" if so.

tapadipti replied:
Order doesn't matter. I'll change it.

tapadipti replied:
Changed.

jorgeorpinel commented on content/docs/studio/user-guide/prepare-your-repositories.md:
```diff
@@ -6,26 +6,15 @@
+models, datasets, metrics and hyperparameters defined in your Git repositories.
+These values can be added to a project in two ways:
+
+1. **Set up DVC in your repositories**: You can use [DVC](https://dvc.org/) and
+   Git to version your code, data and models all within your Git repositories.
+
+   Data Version Control, or DVC, is a data and ML experiment management tool
+   that takes advantage of the existing engineering toolset that you're already
    familiar with (Git, CI/CD, etc.). By using DVC, you can be sure not to bloat
```
Small clarification (less repetitive IMO); And avoid documenting DVC here. Minor (and PR-unrelated)

```suggestion
1. **Set up DVC in your repositories**: You can use [DVC](https://dvc.org/) and
   Git to version your code, data, and models from your project repositories. 
   This avoids bloating
```

jorgeorpinel commented:
Hmmmm the deployment above is going crazy... Please merge `main` but if that doesn't fix it, let's check with @ websites (consider closing the PR temporarily). Thanks

jorgeorpinel commented on content/docs/studio/user-guide/projects-and-experiments/run-experiments.md:
```diff
@@ -79,24 +79,55 @@ on `Commit changes`.
+
+Iterative Studio will create a Git commit with the changes you submitted. This
+commit appears in the project's experiment table. If you had specified a new
+branch to commit the changes to, then a new pull request will also have been
 created from the new branch to the base branch.
```
💅🏼 
```suggestion
branch to commit the changes to, then a new pull request will also be created
from the new branch to the base branch.
```

jorgeorpinel commented on content/docs/studio/user-guide/projects-and-experiments/run-experiments.md:
```diff
@@ -79,24 +79,55 @@ on `Commit changes`.
-CML report icon on mouse hover.
+### Live metrics and plots are tracked
+
+In your model training CI action, you can use [DVCLive] to send live updates to
+metrics and plots back to Iterative Studio, without writing them to your Git
+repository. The live metrics are displayed alongside the corresponding
```
Sorry, I think I asked this before but I forgot the answer (which probably means it's confusing and not sufficiently documented): Where are the metrics stored if not written somewhere in the Git repo? Thanks

jorgeorpinel commented on content/docs/studio/user-guide/projects-and-experiments/run-experiments.md:
```diff
@@ -79,24 +79,55 @@ on `Commit changes`.
-CML report icon on mouse hover.
+### Live metrics and plots are tracked
+
+In your model training CI action, you can use [DVCLive] to send live updates to
+metrics and plots back to Iterative Studio, without writing them to your Git
+repository. The live metrics are displayed alongside the corresponding
+experiment commits. For more details, refer to the
+[next section](/doc/studio/user-guide/projects-and-experiments/live-metrics-and-plots).
```
It's best (shorter, easier to read, evident) to link in-line than create sentences specifically to link. Minor, but this is what I mean:

```suggestion
In your model training CI action, you can use [DVCLive] to [send live updates to
metrics and plots] back to Iterative Studio, without writing them to your Git
repository. The live metrics are displayed alongside the corresponding
experiment commits.

[send live updates to metrics and plots]:
  /doc/studio/user-guide/projects-and-experiments/live-metrics-and-plots
```

jorgeorpinel commented on content/docs/studio/user-guide/projects-and-experiments/live-metrics-and-plots.md:
```diff
@@ -0,0 +1,108 @@
+
+## Set up an access token
+
+Iterative Studio uses access tokens to authorize [DVCLive] to send live updates
+to the metrics and plots.
+
+To set up the access token, open your user profile page. In the , click on
+`Generate new token`.
+
+You can generate, regenerate, or delete your access token from the
+`Studio access token` section of your profile page.
```
This is all one p, I think (sentences about the same thing):
```suggestion
Iterative Studio uses access tokens to authorize [DVCLive] to send live updates
to the metrics and plots.
To set up the access token, open your user profile page. In the , click on
`Generate new token`.
You can generate, regenerate, or delete your access token from the
`Studio access token` section of your profile page.
```

tapadipti replied:
Changed.

jorgeorpinel commented on content/docs/studio/user-guide/projects-and-experiments/live-metrics-and-plots.md:
```diff
@@ -0,0 +1,108 @@
+
+In the training CI job (which has been configured as detailed above), whenever
+you log your metrics or plots using [DVCLive], they will be automatically sent
+to Iterative Studio.
+
+Iterative Studio stores the live metrics and plots data in its database.
```
💅🏼 found another p:
```suggestion
to Iterative Studio. Iterative Studio stores the live metrics and plots
data in its database.
```

Also, this answers my previous question but... do we need more info on what the Studio database is or how live metrics are stored? If it's not relevant to users (not something they can access directly or interact with), then prob not.

tapadipti replied:
Users can't access them directly.

jorgeorpinel commented on content/docs/studio/user-guide/projects-and-experiments/live-metrics-and-plots.md:
```diff
@@ -0,0 +1,108 @@
+2. [Configure your model training CI job](#configure-your-model-training-ci-job)
+3. [Send and view the updates](#send-and-view-live-metrics-and-plots)
+
+https://www.youtube.com/watch?v=hKf4twg832g
+
+TODO: Replace this with a video tutorial for live metrics and plots
```
Here

jorgeorpinel replied:
Still seeing `TODO`s. If they're not planned for this PR let's remove the lines? At least `<!--  -->` them 🙂

tapadipti replied:
Removed.

jorgeorpinel commented on content/docs/studio/user-guide/projects-and-experiments/live-metrics-and-plots.md:
```diff
@@ -0,0 +1,108 @@
+One way to provide the above values to your CI job is to set them as GitHub
+secrets.
+
+Here is an example GitHub action.
+
+TODO: provide link to an example snippet (such as
+[this](https://github.com/iterative/test-dvclive-studio/blob/086a51d76c7983f24c091e1b007820916aa75e7d/.github/workflows/test_live_metrics.yaml#L17-L19))
```
And here

tapadipti replied:
Removed.

jorgeorpinel reviewed (commented):
What's the status on finishing this?

daavoo commented on content/docs/studio/user-guide/projects-and-experiments/live-metrics-and-plots.md:
```diff
@@ -0,0 +1,106 @@
+
+You should define the following environment variables in your CI job:
+
+- `STUDIO_REPO_URL`: URL for the Git repository connected to the project. E.g.,
+  `github:iterative/example-get-started`
```
This is now optional, although we might still want to document it somewhere (stating that by default we will try to get the repo_url automatically) so DVCLive can reference it with a link if/when the automatic way fails.

tapadipti replied:
@daavoo I've completely removed this from the docs. But if there are cases where DVCLive cannot automatically determine the STUDIO_REPO_URL, then we should document those cases and instruct the user to set the environment variable. Can you pls elaborate what such cases could be.

mvshmakov replied:
What should be documented here:
* `STUDIO_REPO_URL` is not needed for the SaaS forges (i.e., when the git repo is stored on these URLs: https://github.com/iterative/dvclive/blob/main/src/dvclive/studio.py#L93-L99).
* `STUDIO_REPO_URL` has a structure of `{remote-type}:{namespace}/{repo-name}`
* For any custom GitLab/GitHub enterprise servers, the users will need to provide the following: `custom-gitlab:iterative/example-get-started` (for custom GitLabs) or `github:iterative/example-get-started` (for GitHub enterprise).


Ideally, we'd like it to be completely removed: https://github.com/iterative/studio/issues/4801. But for this docs iteration, it must be documented.

tapadipti replied:
Done.

shcheklein commented:
@tapadipti @mvshmakov ready to merge this now?

jorgeorpinel commented:
Looks stale

tapadipti commented:
> ready to merge this now?

@mvshmakov has said he'll review it once, and then we can merge it.

jorgeorpinel commented:
Thanks. So is this still a draft? Still merge-blocked? Needs merging `main` branch in, please.

tapadipti commented on content/docs/studio/how-it-works.md:
```diff
@@ -1,48 +1,54 @@
 
 - Using [DVC] and [Git], you will push all your ML experiments to your GitHub,
   GitLab or Bitbucket repositories as Git commits.
 - Using Iterative Studio, or using the command line interface (CLI) of [GTO],
   and possibly [MLEM], you will push all your ML model details to the Git
   repositories as Git commits and Git tags.
```
We don't support other artifacts in Studio. And it is important to mention that the info is saved in Git tags. I'll reword this.

tapadipti replied:
I've reworded it.

tapadipti commented:
I've reviewed and updated the PR, addressed review comments, and resolved the conflicts.
@mvshmakov would you like to review it once or can we merge it now? I'm sending the product update email tomorrow evening, so we should merge this by tomorrow first half (Tuesday morning PT).

mvshmakov commented on content/docs/studio/user-guide/projects-and-experiments/explore-ml-experiments.md:
```diff
@@ -22,21 +21,26 @@ includes metrics, hyperparameters, and information about datasets and models.
   [visualize, compare, and run experiments](#visualize-compare-and-run-experiments).
 
-### Git history
+### Git history and live metrics
```
We also have a special row of "experiment commit", which was initiated by the user and was not yet committed to the repo/errored out. Such rows do not have any metrics but do have modified params/files.

tapadipti replied:
> which was initiated by the user and was not yet committed

The row initiated by the user is committed to Git, right? Do you mean this: `Any live metrics that you send using [DVCLive] are displayed in a special experiment row next to the corresponding Git commit.`

mvshmakov replied:
No, I actually mean information completely unrelated to live metrics. For our "Run experiment" flow, when the user hits "submit experiment", Studio will try to commit this experiment to the repository, but it happens not immediately. Until this experiment is committed (or if it has an error), we display a different type of row which is neither a part of git history, nor a live experiment. It is a special "experiment commit" row type.

I'm wondering whether we should include it here as well for completeness.

tapadipti replied:
Ok. Updated.

mvshmakov commented on content/docs/studio/user-guide/projects-and-experiments/what-is-a-project.md:
```diff
@@ -19,6 +19,9 @@ hyperparameters and models. These values are then presented in an experiment
 table with each experiment (Git commit) in a row and the corresponding values
 for the data, metrics, hyperparameters and models in columns.
 
+Live metrics that you [send to Iterative Studio][live-metrics-and-plots] using
```
Plots?

tapadipti replied:
This is about the project table. The table includes the metrics but not the plots. So I think it is ok.

mvshmakov commented on content/docs/studio/user-guide/projects-and-experiments/live-metrics-and-plots.md:
```diff
@@ -0,0 +1,111 @@
+# Generate live (real-time) metrics and plots for experiments
```
```suggestion
# Generate live (real-time) metrics and plots for remote experiments
```

mvshmakov commented on content/docs/studio/user-guide/projects-and-experiments/live-metrics-and-plots.md:
````diff
@@ -0,0 +1,111 @@
+training code:
+
+```
+from dvclive import Live
+live = Live()
+...
+live.log_metric("accuracy", accuracy)
````
```suggestion
from dvclive import Live

with Live() as live:
  ...
  live.log_metric("accuracy", accuracy)
  live.next_step()
...
live.end()
```

mvshmakov commented on content/docs/studio/user-guide/projects-and-experiments/live-metrics-and-plots.md:
```diff
@@ -0,0 +1,111 @@
+- **Completed** - Iterative Studio does not expect to receive any more updates
+  for these experiments. Once the experiment concludes, you can delete the live
+  metrics row from the project table. Iterative Studio does not automatically
+  push the final results of your experiment to Git. Your CI action should
```
```suggestion
  commit and push the final results of your experiment to Git. Your CI action should
```

mvshmakov reviewed (approved):
Nice work, thanks! A few high-level suggestions:
* Should we rename `live-metrics-and-plots` into `live-experiments` everywhere (in the sidebar, URLs, etc.)?
* Some changes seem to be unrelated to our feature. Please check if the PR is up-to-date before merging it.

tapadipti commented on content/docs/studio/get-started.md:
```diff
@@ -95,11 +95,20 @@ You should now see that a project has been added in your dashboard.
+
+1. You can use your project to [submit new experiments] from Iterative Studio.
+
+2. For the experiments that you submit from Iterative Studio, you can [track
+   live metrics and plots][live-metrics-and-plots] by using [DVCLive].
```
I think `live metrics and plots` is more accurate than `live experiments`. Experiments are always live when they are running; what matters to Studio is that their metrics and plots are displayed in real-time (live).

So I'm not changing this. Pls revert if you feel strongly about this.

tapadipti commented on content/docs/studio/get-started.md:
```diff
@@ -126,26 +135,19 @@ You should now see that a project has been added in your dashboard.
+[submit new experiments]:
+  /doc/studio/user-guide/projects-and-experiments/run-experiments
+[live-metrics-and-plots]:
+  /doc/studio/user-guide/projects-and-experiments/live-metrics-and-plots
```
Same as above. 

> I think `live metrics and plots` is more accurate than `live experiments`. Experiments are always live when they are running; what matters to Studio is that their metrics and plots are displayed in real-time (live).

So I'm not changing this. Pls revert if you feel strongly about this.

tapadipti commented on content/docs/studio/how-it-works.md:
```diff
@@ -55,7 +60,9 @@ https://www.youtube.com/watch?v=5xM5az78Lrg
   upon the creation of Git commits, tags or pull requests. You can use [CML] in
-  your CI/CD actions for continuous machine learning.
+  your CI/CD actions for continuous machine learning. And you can use [DVCLive]
+  to [send live metrics of model training to Iterative
```
Done.

tapadipti commented:
@mvshmakov Thanks for your review. I've addressed your comments. I didn't accept some of your changes. Let me know if you feel strongly about those, and I will update them.

Can you also pls take a quick look at this and confirm: https://github.com/iterative/dvc.org/pull/4012#discussion_r1057286139. Then, I'll update it.

tapadipti commented:
I'm merging this PR now. If you see any more issues, pls create new issue / PR for it.

jorgeorpinel commented on content/docs/studio/user-guide/projects-and-experiments/live-metrics-and-plots.md:
````diff
@@ -0,0 +1,130 @@
+    - If you are using a GitHub enterprise server, set it to
+      `github:iterative/example-get-started`.
+
+## Send and view live metrics and plots
+
+In the training CI job (which has been configured as detailed above), whenever
+you log your metrics or plots using [DVCLive], they will be automatically sent
+to Iterative Studio. Here is an example of how you can use [DVCLive] in your
+training code:
+
+```
+from dvclive import Live
````
Should we also put some of this info in DVCLive docs? It currently doesn't really emphasize the Studio integration.

tapadipti replied:
That would be a good idea.

daavoo replied:
https://github.com/iterative/dvc.org/pull/4306

jorgeorpinel commented on content/docs/studio/user-guide/projects-and-experiments/live-metrics-and-plots.md:
```diff
@@ -0,0 +1,130 @@
+An experiment can have one of the following statuses:
+
+- **Running** - Iterative Studio expects to receive live metrics and plots for
+  these experiments. Note that if the experiment stops due to any error,
+  Iterative Studio will not be aware of this and it will continue to wait for
+  live updates. In this case, you should delete the live metrics row from the
+  project table in Iterative Studio.
```
Another live-related Q @daavoo : couldn't it handle errors and auto live.end() the exp?

daavoo replied:
When using live as a context manager, `with Live() as live`, it does handle (Python) errors and exits with `live.end()`.

For other cases like errors outside Python or when using one of the ML Framework integrations it is not possible

jorgeorpinel commented on content/docs/studio/user-guide/projects-and-experiments/run-experiments.md:
```diff
@@ -146,27 +146,59 @@ automatically be created from the new branch to the base branch. Now, click on
-At this point, the new experiment appears in the project's experiment table. If
-you just committed to a new branch, then a new pull request will also have been
-created from the new branch to the base branch.
+## What happens after you submit a new experiment
+
+Iterative Studio will
+[create the Git commit and pull request](#git-commit-and-pull-request-are-created).
```
This is still somewhat confusing. We're submitting an experiment but a regular Git commit is created?

tapadipti replied:
Yes. The experiment you submit from Studio is not a `dvc exp` experiment. It's a Git commit. We've called it experiment from the beginning, because we expect you to have configured a CI job for model training. We could consider renaming it, but it'll impact many pages in the docs. So, it should become a different issue/PR.

jorgeorpinel commented on content/docs/studio/user-guide/projects-and-experiments/run-experiments.md:
```diff
@@ -146,27 +146,59 @@ automatically be created from the new branch to the base branch. Now, click on
+
+### Git commit (and pull request) are created
+
+Iterative Studio will create a Git commit with the changes you submitted. This
+commit appears in the project's experiment table. If you had specified a new
```
(Same issue:) A regular commit appears as an experiment?

jorgeorpinel replied:
p.s. I guess it's related to calling all these things "rows" both in the UI and docs. If that update is planned, we can wait for the core update to clarify all these texts.

tapadipti replied:
yes

jorgeorpinel commented on content/docs/studio/user-guide/projects-and-experiments/run-experiments.md:
```diff
@@ -146,27 +146,59 @@ automatically be created from the new branch to the base branch. Now, click on
+In your model training CI action, you can use [DVCLive] to [send live updates to
+metrics and plots] back to Iterative Studio, without writing them to your Git
+repository. The live metrics are displayed alongside the corresponding
+experiment commits.
```
"experiment rows" 🙂

tapadipti replied:
👍
